crates/cdk-sqlite/src/wallet/mod.rs

```diff
@@ -734,6 +693,29 @@ FROM proof;
         }
     }
 
+    async fn update_proofs_state(&self, ys: Vec<PublicKey>, state: State) -> Result<(), Self::Err> {
+        let mut transaction = self.pool.begin().await.map_err(Error::from)?;
+
+        for y in ys {
+            sqlx::query(
+                r#"
+UPDATE proof
+SET state=?
+WHERE y IS ?;
+                "#,
+            )
+            .bind(state.to_string())
+            .bind(y.to_bytes().to_vec())
+            .execute(&mut *transaction)
+            .await
+            .map_err(Error::from)?;
+        }
+
+        transaction.commit().await.map_err(Error::from)?;
+
+        Ok(())
+    }
```

Comment:
I know this is just following the bad example I set, but can we remove the for loop and do an in clause like Ive updated the mint to use. 

https://github.com/cashubtc/cdk/blob/72dff953220768db129f5ce7d4c4fb337c6f838c/crates/cdk-sqlite/src/mint/mod.rs#L1152-L1173